nix-darwin: list-keyboards flake output fails to build

This is the error I get:

```
$ darwin-rebuild switch --flake .
...
error: builder for '/nix/store/vfjplmazzm98ak7325a3srldm7vnsh9p-list-keyboards.drv' failed with exit code 2;
       last 15 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/6yhj3nb9hw18cw6nxj0m8aqbzidhs7rz-mac
       > source root is mac
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/5wjj1f3pa1ndi3kvskm1fcs9kdy2xajf-bash-5.2p37/bin/bash
       > clang list-keyboards.c -o list-keyboards -framework IOKit -framework CoreFoundation
       > list-keyboards.c:21:52: error: use of undeclared identifier 'kIOMainPortDefault'
       >    21 |     kern_return_t r = IOServiceGetMatchingServices(kIOMainPortDefault,
       >       |                                                    ^
       > 1 error generated.
       > make: *** [Makefile:10: list-keyboards] Error 1
       For full logs, run 'nix log /nix/store/vfjplmazzm98ak7325a3srldm7vnsh9p-list-keyboards.drv'.
...
$ nix log /nix/store/vfjplmazzm98ak7325a3srldm7vnsh9p-list-keyboards.drv
[same output as above]
```

--- a/c_src/mac/list-keyboards.c
+++ b/c_src/mac/list-keyboards.c
@@ -1,6 +1,12 @@
 #include <IOKit/hid/IOHIDLib.h>
 #include <IOKit/hidsystem/IOHIDShared.h>
 
+/* The name was changed from "Master" to "Main" in Apple SDK 12.0 (Monterey) */
+/* MAC_OS_X_VERSION_12_0 does not exist :( */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 120000
+    #define kIOMainPortDefault kIOMasterPortDefault
+#endif
+
 int main() {
     CFMutableDictionaryRef matching_dictionary = IOServiceMatching(kIOHIDDeviceKey);
     if(!matching_dictionary) {

--- a/changelog.md
+++ b/changelog.md
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0)
 ### Fixed
 
 - Fixed excessive backtracking in the parser leading to unreadable errors (#985)
+- Fixed the `c_src/mac/list-keyboards.c` on Apple SDK < 12.0 (#987)
 
 ## 0.4.4 – 2025-04-11
 

